test/inductor/test_torchinductor.py

```diff
@@ -6784,6 +6786,48 @@ def fn(x: torch.Tensor) -> torch.Tensor:
             out[0].sum().backward()
             self.assertEqual(inp.grad, inp_ref.grad)
 
+        def test_optimize_indexing_assert(self):
+            def fn(x: torch.Tensor) -> torch.Tensor:
+                s = 0.7 * torch.arange(x.shape[0], device=x.device)
+                return x[s.long()]
```

Comment:
Do we have some test cases where the assert isn't removed

Reply:
other than those where we pass bounded tensors from one kernel to another, of course.

Reply:
@lezcano I think it's important to test that bounds are generated in some cases, like the `index_put_` example where you were getting no upper bound generated.

Reply:
Less important, by another good test would be user supplied indices with a clamp call that forces it into range, which happens in `grid_sampler` for example.

Reply:
Added a test for `index_put_`. The `clamp(0)` example does not work yet, I will add a PR to this stack fixing it.